File: src/main/java/mediathek/gui/dialog/reset/ResetSettingsDialog.java
package mediathek.gui.dialog.reset;

import mediathek.gui.dialog.StandardCloseDialog;

import javax.swing.*;
import java.awt.*;

public class ResetSettingsDialog extends StandardCloseDialog {
    public ResetSettingsDialog(Frame owner) {
        super(owner, "Programm zurücksetzen", true);
        setResizable(false);
    }

    @Override
    public JComponent createContentPanel() {
        return new ResetSettingsPanel(null);
    }
}

File: src/main/kotlin/mediathek/gui/dialog/reset/ResetSettingsDialog.kt
package mediathek.gui.dialog.reset

import mediathek.config.Daten
import mediathek.gui.dialog.StandardCloseDialog
import mediathek.mainwindow.SettingsResetHost
import javax.swing.JComponent

class ResetSettingsDialog(
    private val host: SettingsResetHost,
    private val daten: Daten,
) : StandardCloseDialog(host.ownerFrame(), "Programm zurücksetzen", true) {
    init {
        isResizable = false
    }

    override fun createContentPanel(): JComponent = ResetSettingsPanel(host, daten)
}
